src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    public enum Operation {
+        BALANCE("b"), SUPPLY("s"), PURCHASE("p"), RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
+            return Arrays.stream(values())
+                    .filter(op -> op.getCode().equals(code))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException(
+                            "Invalid operation code: " + code)
+                    );
+        }
+    }
+
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
```

Comment:
The constructor is correctly setting the values of the fields. However, it's a good practice to validate the input parameters to ensure they are not null (for `operation` and `fruit`) and that `quantity` is not negative, as these could lead to potential issues later on.